progress.ui/src/org/netbeans/modules/progress/ui/StatusLineComponent.java

```diff
@@ -502,7 +503,18 @@ private void removeListItem(InternalHandle handle) {
         // 2. on mac, needs an owner frame otherwise hiding tooltip also hides the popup. (linux requires no owner frame to force heavyweight)
         // 3. the created window is not focusable window
         if (popupWindow == null) {
-            popupWindow = new JWindow(WindowManager.getDefault().getMainWindow());
+            class ProgressPopUpWindow extends JWindow implements Leafable{
+                ProgressPopUpWindow(Frame parent) {
+                    super(parent);
+                }
+                @Override
+                public boolean isLeaf() {
+                    //never a parent
+                    return true;
+                }
+            }
+            popupWindow = new ProgressPopUpWindow(WindowManager.getDefault().
+                    getMainWindow());
```

Comment:
Wouldn't it be better if this part of the code used `DialogDisplayer` to display its `ProgressPopUpWindow`? Then all the logic can remain in `NbPresenter` and there may be no need for special API.

Reply:
True, but, as far as I know, `DialogDisplayer` doesn't have a way to request an un-decorated window or a JWindow.

Reply:
Can't the code get JDialog and just call [setUndecorated(true)](https://docs.oracle.com/javase/8/docs/api/java/awt/Dialog.html#setUndecorated-boolean-)?